core/types/pow/qitmeer_keccak256.go

```diff
@@ -117,5 +110,13 @@ func (this *QitmeerKeccak256) BlockData() PowBytes {
 
 //check pow is available
 func (this *QitmeerKeccak256) CheckAvailable() bool {
-	return this.params.GetPercentByHeight(this.mainHeight).QitmeerKeccak256Percent > 0
+	return this.params.GetPercentByHeightAndType(this.mainHeight, this.PowType) > 0
```

Comment:
the default checkAvailable behavior can be extract into a common method to remove the duplicated code for all pow types.